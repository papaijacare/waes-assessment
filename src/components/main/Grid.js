import React from 'react';
import { Grid, Segment } from 'semantic-ui-react'

import { ButtonGroup } from '../colors';
import { TextFilter, TextMarker } from "../words";
import { ROLES } from "../../constants";

const COLUMNS_SIZE = 3;

export default () => (
    <Grid divided='vertically'>
      <Grid.Row columns={COLUMNS_SIZE} verticalAlign='middle'>
        <Grid.Column width={2}>
          <Segment circular inverted>Mark!</Segment>
        </Grid.Column>
        <Grid.Column>
          <ButtonGroup role={ROLES.MARKING} />
        </Grid.Column>
        <Grid.Column width={8}>
          <TextMarker />
        </Grid.Column>
      </Grid.Row>

      <Grid.Row columns={COLUMNS_SIZE} verticalAlign='middle'>
        <Grid.Column width={2}>
          <Segment circular inverted>Filter!</Segment>
        </Grid.Column>
        <Grid.Column>
          <ButtonGroup role={ROLES.FILTERING} />
        </Grid.Column>
        <Grid.Column stretched={true}>
          <TextFilter />
        </Grid.Column>
      </Grid.Row>
    </Grid>
);