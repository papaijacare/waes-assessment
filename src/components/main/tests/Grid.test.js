import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Grid from '../Grid'

describe('Grid', () => {
    test('renders correctly', () => {
        const wrapper = shallow(<Grid />)

        expect(toJson(wrapper)).toMatchSnapshot()
    })    
})