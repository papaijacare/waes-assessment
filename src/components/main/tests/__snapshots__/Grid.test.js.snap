// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid renders correctly 1`] = `
<Table
  as="table"
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
      textAlign="center"
    >
      <TableHeaderCell
        as="th"
      />
      <TableHeaderCell
        as="th"
      >
        COLORS
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        WORDS
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <TableRow
      as="tr"
      cellAs="td"
      verticalAlign="top"
    >
      <TableCell
        as="td"
      >
        <Label
          ribbon={true}
          size="big"
        >
          Mark!
        </Label>
      </TableCell>
      <TableCell
        as="td"
      >
        <BlockGroup
          role="ROLE_MARKING"
        />
      </TableCell>
      <TableCell
        as="td"
      >
        <Connect(TextMarker) />
      </TableCell>
    </TableRow>
    <TableRow
      as="tr"
      cellAs="td"
      verticalAlign="top"
    >
      <TableCell
        as="td"
      >
        <Label
          ribbon={true}
          size="big"
        >
          Filter!
        </Label>
      </TableCell>
      <TableCell
        as="td"
      >
        <BlockGroup
          role="ROLE_FILTERING"
        />
      </TableCell>
      <TableCell
        as="td"
      >
        <Connect(TextFilter) />
      </TableCell>
    </TableRow>
  </TableBody>
</Table>
`;
